gui/dig.lua

```diff
@@ -0,0 +1,658 @@
+-- A GUI front-end for the digging designations
+--@ module = false
+
+-- TODOS ====================
+
+-- Must Haves
+-----------------------------
+-- Figure out why pause/unpause doesn't work
+-- Line 'shape' between two points with thickness (currently tricky with how shapes are defined as within bounds rectangle)
+--   A totally vertical line with thickness of e.g. 3 would end up with tiles outside the bounds/view rectangle
+--   Could also allow 'multi-point' line that would produce curves
+-- Need to support classic mode? Might just be double checking non-graphics 'pen' settings
+
+-- Should Haves
+-----------------------------
+-- As the number of shapes and designations grow it might be better to have list menus for them instead of cycle
+-- 3D shapes, would allow stuff like spiral staircases/minecart tracks and other neat stuff, probably not too hard
```

Comment:
this might be more in the realm of quickfort and blueprints. For example, `quickfort run library/dreamfort.csv -n /central_stairs --repeat down,10` will give you 20 levels of spiral staircase. If there is a way to combine the functionality of `gui/dig`, `gui/blueprint`, and `gui/quickfort`, then I'm all ears. I recorded a few notes on the subject here: https://github.com/DFHack/dfhack/issues/2446